reconcile/fleet_labeler/validate.py

```diff
@@ -0,0 +1,66 @@
+from collections.abc import Mapping
+
+from reconcile.gql_definitions.fleet_labeler.fleet_labels import (
+    FleetLabelsSpecV1,
+    OpenShiftClusterManagerV1,
+)
+
+
+class OCMAccessTokenClientIdMissing(Exception):
+    pass
+
+
+class OCMAccessTokenClientSecretMissing(Exception):
+    pass
+
+
+class OCMAccessTokenUrlMissing(Exception):
+    pass
+
+
+class MatchLabelsNotUniqueError(Exception):
+    pass
+
+
+def validate_label_specs(specs: Mapping[str, FleetLabelsSpecV1]) -> None:
+    """
+    We cannot catch all potential errors through json schema definition.
+    """
+    for spec in specs.values():
+        _validate_ocm_token_spec(spec.ocm)
+        _validate_match_labels(spec)
+
+
+def _validate_match_labels(spec: FleetLabelsSpecV1) -> None:
+    """
+    Match labels should be unique within the same spec.
+    """
+
+    seen: set[str] = set()
+    for label_default in spec.label_defaults:
+        key = ""
+        for k, v in sorted(dict(label_default.match_subscription_labels).items()):
+            key += f"{k}:{v},"
+        if key in seen:
+            raise MatchLabelsNotUniqueError(
+                f"The 'matchSubscriptionLabels' combinations must be unique within a spec. Found duplicate: {label_default.match_subscription_labels}"
+            )
+        seen.add(key)
```

Comment:
if you need to gather all duplicates (more than one), can use `Counter`.

Reply:
Using `Counter` now :+1:

Reply:
the new code still only throw for one duplicate, if using `Counter` then you don't have to manually `+= 1` count, get all results like this

```python
    keys = (
        ".".join(
            f"{k}:{v}"
            for k, v in sorted(dict(label_default.match_subscription_labels).items())
        )
        for label_default in spec.label_defaults
    )
    duplicates = [key for key, count in Counter(keys).items() if count > 1]
    if duplicates:
        raise MatchLabelsNotUniqueError(
            f"The 'matchSubscriptionLabels' combinations must be unique within a spec. Found duplicates: {duplicates}"
        )

```

Reply:
Yeah I wanted to print the original json version before the `.join` operation. So I decided to against list comprehension here to keep current var context.

Reply:
Ah ok you want to print all offending label matchers. Good point!

Reply:
Fixed